cmds/shasum/shasum.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"crypto/sha1"
+	"crypto/sha256"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+func helpPrinter() {
+
+	fmt.Printf("Usage:\nshasum -a <algorithm> <File Name>\n")
+	flag.PrintDefaults()
+	os.Exit(0)
+}
+
+func versionPrinter() {
+	fmt.Println("shasum utility, URoot Version.")
+	os.Exit(0)
+}
+
+func getInput(fileName string) (input []byte, err error) {
+
+	if fileName != "" {
+		return ioutil.ReadFile(fileName)
+	}
+	return ioutil.ReadAll(os.Stdin)
+}
+
+//
```

Comment:
Comment should be a complete sentence that begins with the name of the element it describes. This important convention allows generating documentation in a variety of formats. See https://blog.golang.org/godoc-documenting-go-code